### PR TITLE
Add LogRound layer to convert numeric values to integer IDs

### LINKED ISSUE #1910: support LogRound preprocessing layer
We could use implement it based on Discretization layer with a default bin [0, 2, 4, 8, ...]

### PR DESCRIPTION
**Pull request by workingloong**
Fix #1910

### CODE REVIEW

File: elasticdl_preprocessing/layers/log_round.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        **kwargs: Keyword arguments to construct a layer.
+
+    Input shape: A numeric `Tensor`, `SparseTensor` or `RaggedTensor` of shape
+        `[batch_size, d1, ..., dm]`
+
+    Output shape: An int64 tensor of shape `[batch_size, d1, ..., dm]`
+
+    """
+
+    def __init__(self, num_bins, default_value=0, base=None):
```
Comment (brightcoder01) [inline elasticdl_preprocessing/layers/log_round.py:40]: I think we need a `offset` or `bias` parameter.

File: elasticdl_preprocessing/tests/log_round_test.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import tensorflow as tf
+
+from elasticdl_preprocessing.layers import LogRound
+from elasticdl_preprocessing.tests.test_utils import (
+    ragged_tensor_equal,
+    sparse_tensor_equal,
+)
+
+
+class RoundIdentityTest(unittest.TestCase):
```
Comment (brightcoder01) [inline elasticdl_preprocessing/tests/log_round_test.py]: RoundIdentityTest?

File: elasticdl_preprocessing/layers/log_round.py (hunk 1 above)
Comment (workingloong) [reply elasticdl_preprocessing/layers/log_round.py:40]: Can you give an use case to explain why we need it?

Comment (brightcoder01) [reply elasticdl_preprocessing/layers/log_round.py:40]: We will get minus infinite if we pass `0` as input or an exception will be thrown?

Comment (workingloong) [reply elasticdl_preprocessing/layers/log_round.py:40]: If the input is zero or a minus, the output of `tf.math.log` is -inf. Then, `tf.where` will replace it with zero to output.
```python
values = tf.where(
            tf.logical_or(values < 0, values >= num_bins),
            x=tf.fill(dims=tf.shape(values), value=default_value),
            y=values,
        )
```
And I have added the test case with input `0` in the unit test.

Comment (brightcoder01) [reply elasticdl_preprocessing/layers/log_round.py:40]: We can omit the bias parameter at this time. If user needs, we can add it later.

**Review by brightcoder01 — APPROVED** [review]

**Pull request merged**